lib/nxt_support/services/base.rb

```diff
@@ -0,0 +1,37 @@
+module NxtSupport
+  module Services
+    module Base
+      module ClassMethods
+        def class_interface(config = :call)
+          if config.is_a?(Hash)
+            define_singleton_method config.values.first do |*args, **opts|
```

Comment:
```suggestion
            define_singleton_method config.keys.first do |*args, **opts|
```

Reply:
Since we pass something like `class_interface call: :build` (or more technically `class_interface({call: :build})`), we expect `#build` to be defined as a method. If we take `config.keys` though, we would have `call` and not `build`. Or am I not getting it?

Reply:
That's exactly the point of the discussion above 😄 

If I understood correctly, the point of allowing hash is to have a little bit of flexibility. For example,
```ruby
class HouseBuilder
  include NxtSupport::Services::Base

  class_interface call: :build

  def build; end
end

HouseBuilder.call # instead of HouseBuilder.build
```
In this example, `HouseBuilder` has the instance method `#build`, but we want to have the class interface `::call`.
To do so, we need to define the class method `::call` instead of `::build`.

However, with the current implementation, 
```ruby
class_interface :build
```
and
```ruby
class_interface call: :build
```
behave exactly the same - both define the class method `::build` that delegates to the instance method `#build`.

Reply:
@AkihikoITOH I get it now! I think the intention is still to be able to call `HouseBuilder.build` (or any other custom name the service has). Just instead of initializing a new instance (like `HouseBuilder.new.build`).
`HouseBuilder.call` in this case is not that necessary since originally the method name in the service is `build` and it's intended to be used with that naming. But I get the idea and I like it!

On one side we are (currently) able to call `HouseBuilder.build`, or `CarWasher.wash` (`wash` being the main, call-like method here), or whatever instead of `CarWasher.new.wash` etc, but on the other side (your proposal) we would be able to call `.call` on any service class, no matter what the original instance method name is (`call`, `build`, `wash` or whatever)
I think I prefer both, and in case we start using this NxtSupport sugar, what would be the most intuitive option? I think that using `Service.call` while there is only `build` instance method defined would be a little confusing but it's just imho 🤔

To make it less confusing and to not have both similar ways that have the same results I think we can only leave this way of definition:
`class_interface :build`
(as you have proposed initially)
And it will simply define the class method `#build` based on creating an instance and sending the instance method to it. What do you think?

Reply:
@ston1x I think we would be doing either `class_interface :call` or `class_interface :build` in most cases anyway.

I would suggest we stick to the way it was initially done, and if there's a need for the flexibility, we add it at that point🙂

Reply:
@ston1x so should I remove the hash-mapping implementation then and just always exptect Symbols?